Bug: disconnect signature on PySide2

As noticed in a test over in pymmcore-widgets, it's possible that `listeners_connected` is not compatible with older versions of PySide2:

```pytb
E       pytest.PytestUnhandledThreadExceptionWarning: Exception in thread Thread-3
E       
E       Traceback (most recent call last):
E         File "C:\Users\talley\mambaforge\envs\py38\lib\site-packages\pymmcore_plus\_util.py", line 440, in listeners_connected
E           yield
E         File "C:\Users\talley\mambaforge\envs\py38\lib\site-packages\pymmcore_plus\mda\_thread_relay.py", line 104, in mda_listeners_connected
E           ...  # TODO: log reminaing events
E         File "C:\Users\talley\mambaforge\envs\py38\lib\contextlib.py", line 120, in __exit__
E           next(self.gen)
E         File "C:\Users\talley\mambaforge\envs\py38\lib\site-packages\pymmcore_plus\_util.py", line 445, in listeners_connected
E           sig.disconnect(token)
E       TypeError: 'PySide2.QtCore.QObject.disconnect' called with wrong argument types:
E         PySide2.QtCore.QObject.disconnect(QMDASignaler, str, bool)
E       Supported signatures:
E         PySide2.QtCore.QObject.disconnect(PySide2.QtCore.QMetaObject.Connection)
E         PySide2.QtCore.QObject.disconnect(PySide2.QtCore.QObject, bytes, typing.Callable)
E         PySide2.QtCore.QObject.disconnect(bytes, typing.Callable)
E         PySide2.QtCore.QObject.disconnect(PySide2.QtCore.QObject, typing.Union[bytes, NoneType] = None)
E         PySide2.QtCore.QObject.disconnect(bytes, PySide2.QtCore.QObject, bytes)
E         PySide2.QtCore.QObject.disconnect(PySide2.QtCore.QObject, PySide2.QtCore.QMetaMethod, PySide2.QtCore.QObject, PySide2.QtCore.QMetaMethod)
E         PySide2.QtCore.QObject.disconnect(PySide2.QtCore.QObject, bytes, PySide2.QtCore.QObject, bytes)
E       
E       During handling of the above exception, another exception occurred:
E       
E       Traceback (most recent call last):
E         File "C:\Users\talley\mambaforge\envs\py38\lib\threading.py", line 932, in _bootstrap_inner
E           self.run()
E         File "C:\Users\talley\mambaforge\envs\py38\lib\threading.py", line 870, in run
E           self._target(*self._args, **self._kwargs)
E         File "C:\Users\talley\mambaforge\envs\py38\lib\site-packages\pymmcore_plus\mda\_runner.py", line 209, in run
E           self._finish_run(sequence)
E         File "C:\Users\talley\mambaforge\envs\py38\lib\contextlib.py", line 120, in __exit__
E           next(self.gen)
E         File "C:\Users\talley\mambaforge\envs\py38\lib\site-packages\pymmcore_plus\mda\_thread_relay.py", line 104, in mda_listeners_connected
E           ...  # TODO: log reminaing events
E         File "C:\Users\talley\mambaforge\envs\py38\lib\contextlib.py", line 131, in __exit__
E           self.gen.throw(type, value, traceback)
E         File "C:\Users\talley\mambaforge\envs\py38\lib\site-packages\pymmcore_plus\_util.py", line 445, in listeners_connected
E           sig.disconnect(token)
E       TypeError: 'PySide2.QtCore.QObject.disconnect' called with wrong argument types:
E         PySide2.QtCore.QObject.disconnect(QMDASignaler, str, bool)
E       Supported signatures:
E         PySide2.QtCore.QObject.disconnect(PySide2.QtCore.QMetaObject.Connection)
E         PySide2.QtCore.QObject.disconnect(PySide2.QtCore.QObject, bytes, typing.Callable)
E         PySide2.QtCore.QObject.disconnect(bytes, typing.Callable)
E         PySide2.QtCore.QObject.disconnect(PySide2.QtCore.QObject, typing.Union[bytes, NoneType] = None)
E         PySide2.QtCore.QObject.disconnect(bytes, PySide2.QtCore.QObject, bytes)
E         PySide2.QtCore.QObject.disconnect(PySide2.QtCore.QObject, PySide2.QtCore.QMetaMethod, PySide2.QtCore.QObject, PySide2.QtCore.QMetaMethod)
E         PySide2.QtCore.QObject.disconnect(PySide2.QtCore.QObject, bytes, PySide2.QtCore.QObject, bytes)
```

--- a/.github/workflows/test_and_deploy.yml
+++ b/.github/workflows/test_and_deploy.yml
@@ -56,7 +56,8 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -e .[test,${{ matrix.qt }}]
+          pip install -e .[test]
+          pip install ${{ matrix.qt }}
 
       - name: Downgrade pydantic
         if: ${{ matrix.pydantic }}

--- a/.github/workflows/test_and_deploy.yml
+++ b/.github/workflows/test_and_deploy.yml
@@ -27,10 +27,18 @@ jobs:
         platform: [windows-latest, macos-latest]
         python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
         pydantic: [""]
+        qt-backend: ["PySide6"]
         include:
           - platform: windows-latest
             python-version: "3.8"
             pydantic: "'pydantic<2'"
+          - platform: windows-latest
+            python-version: "3.9"
+            pydantic: "'pydantic<2'"
+            qt-backend: "PySide2"
+          - platform: windows-latest
+            python-version: "3.11"
+            qt-backend: "PyQt6"
           - platform: windows-latest
             python-version: "3.11"
             pydantic: "'pydantic<2'"
@@ -48,7 +56,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -e .[test]
+          pip install -e .[test,${{ matrix.qt-backend }}]
 
       - name: Downgrade pydantic
         if: ${{ matrix.pydantic }}
@@ -69,14 +77,14 @@ jobs:
       - name: Build Micro-Manager
         if: runner.os == 'Linux' && steps.cache-mm-build.outputs.cache-hit != 'true'
         run: mmcore build-dev
-        
+
       - name: Install Micro-Manager
         if: runner.os != 'Linux' && steps.cache-mm-build.outputs.cache-hit != 'true'
         run: mmcore install
 
       - name: Remove Qt
         if: runner.os == 'Linux'
-        run: pip uninstall -y pytest-qt qtpy PyQt6 PySide6
+        run: pip uninstall -y pytest-qt qtpy
 
       - name: Test
         run: pytest -v --color=yes --cov=pymmcore_plus --cov-report=xml
@@ -171,7 +179,7 @@ jobs:
         run: |
           python -m pip install -e .[test]
           python -m pip install pytest-codspeed
-          python -m pip uninstall -y pytest-qt qtpy PySide2 PyQt5 PyQt6 PySide6
+          python -m pip uninstall -y pytest-qt qtpy
 
       - name: Cache Linux Build
         id: cache-mm-build

--- a/.github/workflows/test_and_deploy.yml
+++ b/.github/workflows/test_and_deploy.yml
@@ -32,6 +32,7 @@ jobs:
           - platform: windows-latest
             python-version: "3.8"
             pydantic: "'pydantic<2'"
+            qt: "PySide6"
           - platform: windows-latest
             python-version: "3.9"
             pydantic: "'pydantic<2'"
@@ -42,6 +43,7 @@ jobs:
           - platform: windows-latest
             python-version: "3.11"
             pydantic: "'pydantic<2'"
+            qt: "PySide6"
           - platform: ubuntu-latest
             python-version: "3.11"
 
@@ -57,7 +59,10 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -e .[test]
-          pip install ${{ matrix.qt }}
+
+      - name: Install  ${{ matrix.qt }}
+        if: ${{ matrix.qt }}
+        run: pip install ${{ matrix.qt }}
 
       - name: Downgrade pydantic
         if: ${{ matrix.pydantic }}

--- a/.github/workflows/test_and_deploy.yml
+++ b/.github/workflows/test_and_deploy.yml
@@ -37,7 +37,7 @@ jobs:
             python-version: "3.9"
             pydantic: "'pydantic<2'"
             qt: "PySide2"
-          - platform: windows-latest
+          - platform: macos-latest
             python-version: "3.11"
             qt: "PyQt6"
           - platform: windows-latest

--- a/.github/workflows/test_and_deploy.yml
+++ b/.github/workflows/test_and_deploy.yml
@@ -27,7 +27,7 @@ jobs:
         platform: [windows-latest, macos-latest]
         python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
         pydantic: [""]
-        qt-backend: ["PySide6"]
+        qt: ["PySide6"]
         include:
           - platform: windows-latest
             python-version: "3.8"
@@ -35,10 +35,10 @@ jobs:
           - platform: windows-latest
             python-version: "3.9"
             pydantic: "'pydantic<2'"
-            qt-backend: "PySide2"
+            qt: "PySide2"
           - platform: windows-latest
             python-version: "3.11"
-            qt-backend: "PyQt6"
+            qt: "PyQt6"
           - platform: windows-latest
             python-version: "3.11"
             pydantic: "'pydantic<2'"
@@ -56,7 +56,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -e .[test,${{ matrix.qt-backend }}]
+          pip install -e .[test,${{ matrix.qt }}]
 
       - name: Downgrade pydantic
         if: ${{ matrix.pydantic }}

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -54,7 +54,6 @@ cli = ["typer >=0.4.2", "rich >=10.2.0"]
 io = ["tifffile >=2021.6.14", "zarr >=2.2"]
 test = [
     "msgpack",
-    "PySide6",
     "pytest-cov >=4",
     "pytest-qt >=4",
     "pytest >=7.3.2",

--- a/src/pymmcore_plus/_util.py
+++ b/src/pymmcore_plus/_util.py
@@ -440,9 +440,13 @@ def listeners_connected(
 
                         ctype = getattr(Qt.ConnectionType, qt_connection_type)
                         token = signal.connect(slot, ctype)  # type: ignore
-                        tokens[attr_name].add(token)
                     else:
-                        tokens[attr_name].add(signal.connect(slot))
+                        token = signal.connect(slot)
+
+                    # This only seems to happen on PySide2
+                    if token is None or isinstance(token, bool):
+                        token = slot
+                    tokens[attr_name].add(token)
 
     try:
         yield

